test/e2e/e2e_test.go

```diff
@@ -387,15 +389,82 @@ func TestNamespaceConstraintsExclude(t *testing.T) {
+
+	clientSet, nodeInformer, stopCh := initializeClient(t)
+	defer close(stopCh)
+
+	// The kube-dns pods have a system-critical priority, but are not daemonsets, static pods, have local storage, etc.
+	namespace := "kube-system"
+	podList, err := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(map[string]string{"k8s-app": "kube-dns"}).String()})
+	if err != nil {
+		t.Fatalf("Unable to list pods: %v", err)
+	}
+
+	initialPodNames := getPodNames(podList.Items)
+	sort.Strings(initialPodNames)
+	t.Logf("Existing pods: %v", initialPodNames)
+
+	t.Logf("set the strategy to delete pods of any priority from %v namespace", namespace)
+	runPodLifetimeStrategy(ctx, clientSet, nodeInformer, &deschedulerapi.Namespaces{
+		Include: []string{namespace},
+	}, "", nil, true, nil)
+
+	// All pods are supposed to be deleted, wait until all the old pods are deleted
+	if err := wait.PollImmediate(time.Second, 20*time.Second, func() (bool, error) {
+		podList, err := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(map[string]string{"k8s-app": "kube-dns"}).String()})
+		if err != nil {
+			return false, nil
+		}
+		includePodNames := getPodNames(podList.Items)
+		// validate all pod were deleted
+		if len(intersectStrings(initialPodNames, includePodNames)) > 0 {
+			t.Logf("Waiting until %v pods get deleted", intersectStrings(initialPodNames, includePodNames))
+			// check if there's at least one pod not in Terminating state
+			for _, pod := range podList.Items {
+				// In case podList contains newly created pods
+				if len(intersectStrings(initialPodNames, []string{pod.Name})) == 0 {
+					continue
+				}
+				if pod.DeletionTimestamp == nil {
+					t.Logf("Pod %v not in terminating state", pod.Name)
+					return false, nil
+				}
+			}
+			t.Logf("All %v pods are terminating", intersectStrings(initialPodNames, includePodNames))
+		}
+
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Error waiting for pods to be deleted: %v", err)
+	}
+}
+
 func TestThresholdPriority(t *testing.T) {
-	testPriority(t, false)
+	// Without this, there's a chance the descheduler strategy will
+	// run while a few pods are in a Pending state. This causes the test case
+	// to fail because the pods will be scheduled after the descheduler has been run
+	t.Logf("Waiting 15s for terminating pods to disappear")
+	time.Sleep(15 * time.Second)
+	testPriority(t, false, false)
 }
 
 func TestThresholdPriorityClass(t *testing.T) {
-	testPriority(t, true)
+	testPriority(t, true, false)
+}
+
+func TestThresholdPriorityDisable(t *testing.T) {
+	t.Logf("Waiting 15s for terminating pods to disappear")
+	time.Sleep(15 * time.Second)
+	testPriority(t, false, true)
+}
+
+func TestThresholdPriorityClassDisable(t *testing.T) {
+	testPriority(t, true, true)
 }
 
-func testPriority(t *testing.T, isPriorityClass bool) {
+func testPriority(t *testing.T, isPriorityClass bool, evictCritical bool) {
```

Comment:
Instead of extending `testPriority` with `evictCritical`, can you add low+high priority pods into `TestEvictSystemCritical` as well? I.e. creating RC with low/high pods in kube-system ns and just evicting all of them alongside `kube-dns` pods.

Reply:
Sure, I can move it.